### DISCUSSION TITLE
Please give an example of Republish

### DISCUSSION THREAD

**Discussion by VuongUranus** [nats-io/nats-server#5615] [General]
I have two questions: how are core NATS subscribers and consumers in jetstream different? And an example of using Subscriber and Republish to increase performance. Thanks a lot.

**Comment by alberk8** [comment]
As far as I know the Publish and Subscribe a fire and forget  where as the Jetstream publish is persistence and the consumer can be set with specific  Acknowledgement policy, Max Redelivery/Retry, Backoff/Retry Timing, etc.

**Comment by ripienaar** [comment]
Republish means the stream sends the message to a normal core subject after receiving it and in headers you will find detsils like the stream sequence. 

Setting it up is just putting a subject in the configuration item in the stream republish field. 

You can use it to avoid making consumers of all you want is the most recent messages on a stream - unfiltered. 

Use the headers to detect if you missed any messages and use a stream get to fetch them.

**Comment by VuongUranus** [comment]
It would be great if I had an example of it. Thanks a lot.

**Reply by ripienaar** [reply]
Let us know what you tried and what happened. As I said. It’s just a subject for where you wish to have the messages published set on the stream configuration.

**Reply by VuongUranus** [reply]
I don't understand why I don't use the NATS Core subscription to listen to messages on that stream but instead use Republish to the NATS Core destination.

**Reply by ripienaar** [reply]
Because if you listen on the republish you know the stream sequences etc.  Core is lossy meaning you can miss messages so knowing the sequence means you can know if you missed something and can go get that missing message directly.

**Reply by VuongUranus** [reply]
I understand the problem, thank you very much. Your team has built a great project with good features.
